lib/rubocop/cop/lint/binary_operator_identical_operands.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for places where binary operator has identical operands.
+      #
+      # This cop is marked as unsafe as it does not consider side effects when calling methods
+      # and thus can generate false positives:
+      #   if wr.take_char == '\0' && wr.take_char == '\0'
+      #
+      # @example
+      #   # bad
+      #   x.top >= x.top
+      #
+      #   if a.x != 0 && a.x != 0
+      #     do_something
+      #   end
+      #
+      #   def childs?
+      #     left_child || left_child
+      #   end
+      #
+      class BinaryOperatorIdenticalOperands < Base
+        MSG = 'Binary operator has identical operands.'
```

Comment:
I'd mention in the message the name of the binary operator (and maybe even the operand in question).

Reply:
Added mention about operator. I think, mentioning the operand is not necessary, as it will be obvious from code. Example: `foo == foo`.